File: go/arrays-and-hashing/palindrome_number.go
package arrays_and_hashing

import "strconv"

func IsPalindrome(x int) bool {

	if x < 0 {
		return false
	}

	if x%10 == 0 && x != 0 {
		return false
	}

	numStr := strconv.Itoa(x)

	left := 0
	right := len(numStr) - 1

	for left < right {
		if numStr[left] != numStr[right] {
			return false
		}

		left++
		right--
	}
	return true
}

File: typescript/arrays-and-hashing/palindrome-number.ts
export function isPalindrome(x: number): boolean {
    if (x < 0) {
        return false;
    }

    if (x % 10 === 0 && x != 0) {
        return false;
    }

    const numStr = x.toString();

    let left = 0;
    let right = numStr.length - 1;

    while (left < right) {
        if (numStr[left] !== numStr[right]) {
            return false;
        }

        left++;
        right--;
    }


    return true;
}